*2016-03-14  Paul Eggert  <eggert@cs.ucla.edu>
*
*	select: try to port to 2016.1.150 compiler
*	Problem reported by Balázs Hajgató in:
*	http://lists.gnu.org/archive/html/bug-gnulib/2016-03/msg00026.html
*	* lib/sys_select.in.h (select): Use 'restrict' on arguments where
*	POSIX specifies 'restrict'.
*
diff --git a/lib/sys_select.in.h b/lib/sys_select.in.h
index a813f9c..d6d3f9f 100644
--- octave-4.0.0/libgnu/sys_select.in.h.orig
+++ octave-4.0.0/libgnu/sys_select.in.h
@@ -289,12 +289,15 @@ _GL_WARN_ON_USE (pselect, "pselect is not portable - "
 #   define select rpl_select
 #  endif
 _GL_FUNCDECL_RPL (select, int,
-                  (int, fd_set *, fd_set *, fd_set *, struct timeval *));
+                  (int, fd_set *restrict, fd_set *restrict, fd_set *restrict,
+                   struct timeval *restrict));
 _GL_CXXALIAS_RPL (select, int,
-                  (int, fd_set *, fd_set *, fd_set *, struct timeval *));
+                  (int, fd_set *restrict, fd_set *restrict, fd_set *restrict,
+                   struct timeval *restrict));
 # else
 _GL_CXXALIAS_SYS (select, int,
-                  (int, fd_set *, fd_set *, fd_set *, struct timeval *));
+                  (int, fd_set *restrict, fd_set *restrict, fd_set *restrict,
+                   struct timeval *restrict));
 # endif
 _GL_CXXALIASWARN (select);
 #elif @HAVE_WINSOCK2_H@
